main.tf

```diff
@@ -147,11 +147,68 @@ resource "metal_device" "esxi_hosts" {
   }
 }
 
+resource "null_resource" "reboot_pre_upgrade" {
+
+  depends_on = [metal_device.esxi_hosts]
+  count      = var.update_esxi ? 1 : 0
+
+  provisioner "local-exec" {
+    command = "sleep 250"
+  }
+}
+
+data "template_file" "upgrade_script" {
```

Comment:
The `templatefile` function is preferred:

https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file